pkg/resources/warehouse.go

```diff
@@ -0,0 +1,150 @@
+package resources
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func Warehouse() *schema.Resource {
+	d := NewResourceWarehouse()
+	return &schema.Resource{
+		Create: d.Create,
+		Read:   d.Read,
+		Delete: d.Delete,
+		Update: d.Update,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: ValidateWarehouseName,
+			},
+			"comment": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+				// TODO validation
+			},
+		},
+	}
+}
+
+type warehouse struct{}
+
+func NewResourceWarehouse() *warehouse {
+	return &warehouse{}
+}
+
+func ValidateWarehouseName(val interface{}, key string) ([]string, []error) {
+	return snowflake.ValidateIdentifier(val)
+}
+
+func (w *warehouse) Create(data *schema.ResourceData, meta interface{}) error {
+	name := data.Get("name").(string)
+	comment := data.Get("comment").(string)
+	db := meta.(*sql.DB)
+
+	err := DBExec(db, "CREATE WAREHOUSE %s COMMENT='%s", name, comment)
```

Comment:
this might be missing a `'` around the comment

Reply:
indeed